### PR TITLE
Edit position command

### PR DESCRIPTION
**Pull request by sueann-chua**
-  Added edit_p command
- Added editpositioncommandparser
- Added EditPositionDescriptionBuilder
- Added PositionBuilder
- Added editpositioncommandparsertest
- Added editpositioncommandtest
- Abstract out position.commandtestutil
- Abstract out position.util
- Added editpositiondescriptortest file
- Modify GUI for positions in position list
- Modify GUI for positions in candidate list

Having some difficulty updating the position field in the candidates.json file. Will publish PR after resolving this issue.

### CODE REVIEW

File: src/test/java/seedu/address/logic/position/CommandTestUtil.java [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+    /**
+     * Convenience wrapper to {@link #assertCommandSuccess(Command, Model, CommandResult, Model)}
+     * that takes a string {@code expectedMessage}.
+     */
+    public static void assertCommandSuccess(Command command, Model actualModel, String expectedMessage,
+                                            Model expectedModel) {
+        System.out.println(command);
+        System.out.println(actualModel);
+        System.out.println(expectedMessage);
+        System.out.println(expectedModel);
```
Comment (wanyu-l) [inline src/test/java/seedu/address/logic/position/CommandTestUtil.java:82]: Perhaps they can be removed?

File: src/test/java/seedu/address/logic/position/CommandTestUtil.java [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+    /**
+     * Executes the given {@code command}, confirms that <br>
+     * - a {@code CommandException} is thrown <br>
+     * - the CommandException message matches {@code expectedMessage} <br>
+     * - the address book, filtered person list and selected person in {@code actualModel} remain unchanged
+     */
+    public static void assertCommandFailure(Command command, Model actualModel, String expectedMessage) {
+        // we are unable to defensively copy the model for comparison later, so we can
+        // only do so by copying its components.
+        HrManager expectedAddressBook = new HrManager(actualModel.getHrManager());
```
Comment (wanyu-l) [inline src/test/java/seedu/address/logic/position/CommandTestUtil.java:96]: Perhaps "expectedAddressBook" can be changed?

File: src/test/java/seedu/address/logic/position/CommandTestUtil.java [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+        List<Person> expectedFilteredList = new ArrayList<>(actualModel.getFilteredPersonList());
+
+        assertThrows(CommandException.class, expectedMessage, () -> command.execute(actualModel));
+        assertEquals(expectedAddressBook, actualModel.getHrManager());
+        assertEquals(expectedFilteredList, actualModel.getFilteredPersonList());
```
Comment (wanyu-l) [inline src/test/java/seedu/address/logic/position/CommandTestUtil.java:101]: Perhaps it should be "Position" instead?

**Review by seaweediman — APPROVED** [review]
LGTM

**Pull request merged**